### PR TITLE
features/clean ups

### PR DESCRIPTION
**Pull request by iamjpn**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -484,33 +490,34 @@ static bool parse_options(int argc, char *argv[])
...
 	}
 
-	if (l_count) {
-		int pir = -1, i, chip, core, thread;
+	return true;
+}
 
-		if (!device_node)
-			return false;
+bool cpus_parse(int args[], int arg_count)
```
Comment (amitay) [inline src/main.c:503]: Please use `static` declaration for all local functions.

**Review by amitay — CHANGES_REQUESTED** [review]
First patch requires a bit of re-work.

File: src/main.c [hunk 2]
```diff
@@ -582,6 +589,11 @@ int main(int argc, char *argv[])
 	if (!pdbg_targets_init(NULL))
 		return 1;
 
+	if (l_count) {
+		if (!cpus_parse(l_list, l_count))
+			return 1;
+	}
+
```
[the remark this answers is not in the document]
Comment (iamjpn) [reply src/main.c:603]: `pir_map()` now uses `pdbg_get_proc()`. So we need to have to have `pdbg_proc` set before we do `cpus_parse()`.
`parse_options()` is called before `pdbg_targets_init()`, etc so at that time `pdbg_proc` is not finalized.
That is why it had to be moved out of `parse_options()` into `main()`.

**Comment by amitay** [comment]
Merged.

**Pull request closed without merging**